File: FC.App/Scripts/Shared/ServiceMessages/FestivalFilter.ts
module FC.Shared.ServiceMessages {
    import CM = FC.Core.CoreModel;
    import INT = FC.Shared.Interfaces;
    import MODELS = FC.Shared.Models;
    import MODULES = FC.Modules;
    export class FestivalFilter implements INT.IFestivalFilter {
        public GenreIDs: Array<string>;
        public CountryIDs: Array<string>;
        public FestivalID: string;
        public MonthNum: number;   
        public YearNum: number; 
    }
}

File: FC.Shared/ServerMessages/FestivalFilter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FC.Shared.ServerMessages
{
    public class FestivalFilter
    {
        public FestivalFilter()
        {
            GenreIDs = new List<Guid?>();
            CountryIDs = new List<Guid?>();
            ArtistIDs = new List<Guid?>();
            LocationIDs = new List<Guid?>();
            FestivalIDs = new List<Guid?>();
        }
        
        public List<Guid?> GenreIDs { get; set; }
        public List<Guid?> CountryIDs { get; set; }
        public List<Guid?> ArtistIDs { get; set; }
        public List<Guid?> LocationIDs { get; set; }
        public List<Guid?> FestivalIDs { get; set; }
        public string FestivalID { get; set; }
        public int MonthNum = -1;
        public int YearNum = DateTime.Now.Year;
        public int PageLength { get; set; }
        public int CurrentLength { get; set; }
        public int BlockLength { get; set; }

        public Guid? LastID { get; set; }
    }
}
